### PR TITLE
feat(gen): Support creating templates without having to pass a context

### PR DESCRIPTION
**Pull request by abdusco**
Closes: #504

### CODE REVIEW

**Comment by abdusco** [comment]
Please don't merge right away, I'll update README + CHANGELOG after you OK this.

**Comment by abdusco** [comment]
Ah yes, I had `no_tests: true` in my config.

File: gen/templates/factory/bobfactory_main.bob_test.go.tpl [hunk 1]
```diff
@@ -22,7 +22,7 @@ func TestCreate{{$tAlias.UpSingular}}(t *testing.T) {
     }
   }()
 
-  if _, err := New().New{{$tAlias.UpSingular}}(ctx).Create(ctx, tx); err != nil {
+  if _, err := New().New{{$tAlias.UpSingular}}WithContext(ctx).Create(ctx, tx); err != nil {
```
Comment (abdusco) [inline gen/templates/factory/bobfactory_main.bob_test.go.tpl:25]: Should the tests call `.New` or `.NewWithContext`? I think the former. Even though a ctx is available, passing it to the builder feels redundant.

Comment (stephenafamo) [reply gen/templates/factory/bobfactory_main.bob_test.go.tpl:25]: If a context is available, let's use it.

Actually, since Bob targets Go 1.24 at minimum, we should use [`t.Context()`](https://pkg.go.dev/testing#T.Context) instead of `context.Background()`

Comment (abdusco) [reply gen/templates/factory/bobfactory_main.bob_test.go.tpl:25]: Alright, I'll change `context.Background()` to `t.Context()` too.

**Comment by abdusco** [comment]
I had forgotten to add a test case unicode enum values in #502 

I squeezed in another test here. I hope you don't mind it.

https://github.com/stephenafamo/bob/pull/506/commits/5dec281e19106d2f7dc516b528767a08157cee02

File: CHANGELOG.md [hunk 2]
```diff
@@ -29,6 +29,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Updated documentation for readability, added code gen examples. (thanks @singhsays)
 - Columns are now matched in a case-insensitive manner in type replacements. (thanks @abdusco)
 - Columns can now be matched with as many conditions as needed in type replacements. This removes the previous requirement that boolean fields had to be specified in addition to a string field. (thanks @abdusco)
+- Factories now expose a `NewXWithContext` that accepts a context in addition to `NewX` that does not. This provides a cleaner API for the callers, while still allowing the use of context internally. (thanks @abdusco)
```
Comment (abdusco) [inline CHANGELOG.md:32]: Turns out the docs never mentioned the use of ctx

https://github.com/stephenafamo/bob/blob/2f1f83bba3c103d5988f3b4774384aaf9986e86d/website/docs/code-generation/factories.md?plain=1#L24

**Comment by abdusco** [comment]
Turns out the docs never mentioned the use of ctx, so I won't touch it for now.

https://github.com/stephenafamo/bob/blob/2f1f83bba3c103d5988f3b4774384aaf9986e86d/website/docs/code-generation/factories.md?plain=1#L24
